alacritty/src/display/mod.rs

```diff
@@ -980,74 +1020,203 @@ impl Display {
     }
 
     /// Format search regex to account for the cursor and fullwidth characters.
-    fn format_search(size_info: &SizeInfo, search_regex: &str, search_label: &str) -> String {
-        // Add spacers for wide chars.
-        let mut formatted_regex = String::with_capacity(search_regex.len());
-        for c in search_regex.chars() {
-            formatted_regex.push(c);
-            if c.width() == Some(2) {
-                formatted_regex.push(' ');
+    fn format_search(search_regex: &str, search_label: &str, max_width: usize) -> String {
+        let label_len = search_label.len();
+
+        // If the label doesn't fit into screen return.
+        if label_len > max_width {
+            return search_label[..max_width].to_owned();
+        }
```

Comment:
This comment makes it sound like we don't render the label at all if it doesn't fit, should probably change it to something like this:

```
// Skip `search_regex` formatting if only label is visible.
```